### ISSUE TITLE
Failing to ``getfield(Main, ...)``

### ISSUE THREAD

**Issue by joa-quim** [tlienart/Franklin.jl#989]
I have several of these errors when building [GMTjl_doc](https://github.com/GenericMappingTools/GMTjl_doc) via GH actions. I **do not** get them when running the same commands locally (both Windows & Linux)

```
evaluating code [example_figure] in (examples\plotting_functions\05_1_stats   
+ Franklin Warning: in <examples\plotting_functions\05_1_stats.md>
¦ There was an error of type 'UndefVarError' when running a code block.
¦ Checking the output files 'D:\a\GMTjl_doc\GMTjl_doc\__site\assets\examples\plotting_functions\05_1_stats\code\output\example_figure.(out|res)'
¦ might be helpful to understand and solve the issue.
¦ 
¦ Relevant pointers:
¦ - evaluation of Julia code blocks: https://franklinjl.org/code/
¦ 
¦ Details:
¦ UndefVarError: scatter! not defined
¦ Stacktrace:
¦  [1] add_opt_module(d::Dict{Symbol, Any})
¦    @ GMT C:\Users\runneradmin\.julia\packages\GMT\M9Rtt\src\common_options.jl:2557
```

The error seems to come from [this line](https://github.com/GenericMappingTools/GMT.jl/blob/master/src/common_options.jl#L2557) 
```
fn = getfield(Main, Symbol(string(this_symb, "!")))
```
where it errors saying that functions that **do exist and and exported** are *not defined*

**Comment by tlienart** [comment]
Hmm yes that's going to be hard to debug/fix. Do the `scatter` (without `!`) work? The `getfield(Main...)` is probably not ideal because it makes assumptions about what `Main` is but that's not going to help you much right now.

In the meantime, a relatively easy workaround would be to generate the website locally since that seems to work fine, call `optimize` etc so the `prepath` is specified correctly, and then push the content of `__site` to `gh-pages`. It adds a manual step but at least will get your site up and going.

Note: I'm aware the way code is executed in Franklin is not super robust, this (among other things) has motivated an in-depth rewrite of the code in the form of Xranklin.jl [1] so I'd be curious to see if it works with it but fair enough if you don't have time to give it a shot

([1] it's the next version of Franklin which will be merged in some future and is available now if you want to try it out at Xranklin.jl)

**Comment by joa-quim** [comment]
Yes, the ``scatter!`` [exists ](https://github.com/GenericMappingTools/GMT.jl/blob/master/src/GMT.jl#L140) and works fine when called directly. It's when it (and others like it) is called nested (example ``image(G, scatter=(...))`` that it takes the ``!`` form and fails when run from Franklin, but only when the run is remote. Locally it works fine too. This is the mysterious part.

I already maintain to locations for the docs. For the first one I copy ``_site`` to the site (ignore the certificate warnings. Something to fix  but it's so boring to learn how-to),

https://fct-gmt.ualg.pt/GMTjl_doc/examples/plotting_functions/05_1_stats/#example_17927783038414841698

and the second is what the GH action do. Note that this second version has several failed commands that show up fine is the ``fct-gmt.ualg.pt``

https://www.generic-mapping-tools.org/GMTjl_doc/examples/plotting_functions/05_1_stats/#example_2653022517396184860

So you are saying that with ``optimize`` and ``publish`` I could set up a solution where I manually could upload to the ``gh-pages``? That would be fine and I actually already tried to do so but failed miserably.

I also gave a look at ``Xranklin.jl`` but because the activity on it has not been much these last months I didn't go to the point of trying it. Will find some time to do so.

Thanks.

**Comment by joa-quim** [comment]
>>   So you are saying that with optimize and publish I could set up a solution where I manually could upload to the gh-pages? That would be fine and I actually already tried to do so but failed miserably.

> yes that's right

Sorry to resurrect this, but how do I do that?

Don't know what is happening but nothing works anymore on Gh-Actions side anymore. Creating on Mac permanently hangs or crashes, on Linux there is a bunch of crazy errors, on Windows it doesn't work because of the ``rsync`` shit. And honestly I'm lacking courage (and time) to dive in Xranklin (it took me a lot of tame to get a working Franklin solution).

Thanks

**Comment by joa-quim** [comment]
Thanks a lot. At least I have something that works now. The option 1. worked but since ``__site`` is not under version control I had to manually ``git add`` many files (mostly figures) that were otherwise missing. But a problem I envisage with this is that since I'm also committing locally my ``.git`` directory will grow enormously.

I would certainly love to use GA only but don't understand what happens in that land. If I use a Linux builder there are tons of incomprehensible errors. Using the MacOS is a lottery where most of the times it hangs or crashes. Only yesterday I was able to have a almost decent build. Windows works fine but that damn ``JamesIves/github-pages-deploy-action@releases/v4`` action is broken because it generates a filename with Windows path and the whole process really dies on the beach. I don't understand their pointer to the alternative solution https://github.com/JamesIves/github-pages-deploy-action#operating-system-support-
Tried it but my ``deploy.yml`` must had some (silent) errors because the build process didn't even start.

**Comment by joa-quim** [comment]
Seems easy to adapt to my case. I'll try it next time I will need to update the manual.
Thanks for all the work it dedicated to this.

**Comment by joa-quim** [comment]
This thing looks cursed.

https://github.com/joa-quim/GMTjl_doc/actions/runs/3955082305/jobs/6773282818

 Now fails again near the very end and with not explanation. But I see an error further up that I don't understand.
https://github.com/joa-quim/GMTjl_doc/actions/runs/3955082305/jobs/6773282818#step:8:718

**Comment by joa-quim** [comment]
You already are. That's the repo I kept for experiences like this. Tried also i the official one and got exactly the same error

https://github.com/GenericMappingTools/GMTjl_doc/actions/runs/3958861369/jobs/6781704426
